docarray/array/array/io.py

```diff
@@ -291,6 +296,69 @@ def to_json(self) -> str:
         """
         return json.dumps([doc.json() for doc in self])
 
+    @classmethod
+    def from_csv(cls, file_path: str, encoding: str = 'utf-8') -> 'DocumentArray':
+        """
+        Load a DocumentArray from a csv file following the schema defined in the
+        :attr:`~docarray.DocumentArray.document_type` attribute.
+        The column names have to match the fields of the document type.
+        For nested fields use dot-separated access paths, such as 'image__url'.
```

Comment:
Can we handle DocumentArray (list-like) fields? I'm guessing not, does it even make sense in csv? either way, I'd mention it here